spec/unit/sensu_check_spec.rb

```diff
@@ -114,4 +114,27 @@
       end
     end
   end
+
+  describe 'ttl_status parameter' do
+    subject { described_class.new(resource_hash)[:ttl_status] }
+
+    valid = [1, 0, '1', '0']
```

Comment:
`'absent'` is also valid and so is `undef`

Reply:
Data type: 

```puppet
Variant[Undef,Enum['absent'],Integer]
```

Reply:
'1' and '0' should not work since they are not integers

Reply:
They work because the type property munges with `to_i` if not absent.  The validation in type takes place before munge.  The data type isn't actually tested yet, should the behavior of type property match data type or should data type be updated to allow numeric strings?

Reply:
Seems the data type should be updated to allow for stringified integers by using a regex.